packages/core/src/operations/terminal-defer.spec.ts

```diff
@@ -0,0 +1,47 @@
+import cuillere, { Cuillere, defer, terminal } from '..'
+
+describe('terminal defer', () => {
+  let cllr: Cuillere
+
+  beforeEach(() => {
+    cllr = cuillere()
+  })
+
+  it('should allow to defer in parent frame', async () => {
+    const actions: string[] = []
+
+    function* resource(id: number) {
+      actions.push(`create resource ${id}`)
+
+      yield terminal(defer(function* cleanupResource() {
+        actions.push(`cleanup resource ${id}`)
+      }))
+    }
```

Comment:
Is it possible to return something ? For a resource it could be very usefull

Reply:
Yes I'll change the test